DVCodingStyle.md

```diff
@@ -0,0 +1,863 @@
+| Agent level config object         | `rand <dut>_agent_cfg m_<dut>_agent_cfg;`    |
+| Agent coverage collection object  | `<dut>_agent_cov cov;`                       |
+| Driver                            | `<dut>_driver driver;`                       |
+| Monitor                           | `<dut>_monitor monitor;`                     |
+| Scoreboard                        | `<dut>_scoreboard scoreboard;`               |
+| Virtual sequencer                 | `<dut>_virtual_sequencer virtual_sequencer;` |
+| Sequencer                         | `<dut>_sequencer <dut>_sequencer;`           |
+
+Additional naming guidelines:
+
+1.  The top level testbench file that instantiates the DUT must be named `tb.sv`.
+    The testbench module must be named `tb`, and the instance name of the DUT in
+    `tb.sv` must be `dut`. Using this standardized naming allows universal usage
+    of any supporting DV sources (coverage cfg files, and so on).
+2.  When instantiating any objects, the `name` argument to `type_id::create()`
+    must match the name of the variable being assigned to.
+    This ensures that any reporting messages refer to the right objects.
+3.  To avoid name collisions, it is recommended that global types and types defined
+    within a package have unique prefixes. This includes typedefs, enumerated
+    types/values, package parameters, package localparams, any functions/tasks, and
+    macros. This prefixing is not necessary for any types defined within a class,
+    module, or interface, as they will be scoped by the parent entity.
+4.  Note that all package variables (parameters, etc.) must have a specified type.
+5.  When instantiating a user-defined class member variable that is a class object:
+    *   Prefix the instance name of the class object with `m_`, as such:
+
+        ```systemverilog
+        my_animal m_my_animal;
+        ```
+
+        This scheme does not apply to any class member variables that are
+        not themselves class objects.
+
+    *   If multiple instantiations of the class are desired without an array
+        to store them, use `<class_name> m_<instance_name>` to declare a variable,
+        and use an informative instance name.
+
+        ```systemverilog
+        my_animal m_my_cat;
+        my_animal m_my_dog;
+        my_animal m_my_fish;
+        ```
+
+    *   If an array of class handles is desired, choose a name such that it follows
+        one of the following schemes:
+
+        ```systemverilog
+        int NUM_ANIMALS = 10;
+        // This plural convention is recommended, but both are equally acceptable
+        my_animal m_my_animals[NUM_ANIMALS];
+        my_animal m_my_animal_array[NUM_ANIMALS];
+        ```
+
+
+## Directory Structure and Packages
+
+Define one class per file, and the filename should match the name of the class.
+The class files must be `` `include``-ed into the package file.
+Directory structure of the `uvm_agent` and `uvm_env` testbench components
```

Comment:
nit: UVM agent and UVM environment instead of `uvm_agent` and `uvm_env`

Reply:
Fixed